### PR TITLE
zqso robust to missing *_SCND_TARGET columns; record quasarnp version

### PR DESCRIPTION
**Pull request by sbailey**
This PR fixes a `make_zqso_files` bug when the input file FIBERMAP is missing a `*_SCND_TARGET` input column (because the input fiberassign file was also missing that column).  e.g. from the example in PR 737:
```
[cori06 ~] make_zqso_files -t 80605 -n 20210205 -o $CSCRATCH/desi/zqsotest
INFO:lyazcat.py:66:tmark: 
    Loading QuasarNP Model file and lines of interest: 2021-06-03 | 14:58:14
INFO:lyazcat.py:66:tmark: 
      QNP model file loaded: 2021-06-03 | 14:58:14
INFO:make_zqso_files:141:<module>: processing TILEID=80605, NIGHTID=20210205, petal=0
INFO:lyazcat.py:66:tmark: 
    Making redrock zcat: 2021-06-03 | 14:58:14
Traceback (most recent call last):
  File "/global/common/software/desi/cori/desiconda/20200801-1.4.0-spec/code/desitarget/master/bin/make_zqso_files", line 143, in <module>
    create_zcat(input_dir, output_dir, tile=tile,
  File "/global/common/software/desi/cori/desiconda/20200801-1.4.0-spec/code/desitarget/master/py/desitarget/lyazcat.py", line 754, in create_zcat
    zcat = make_new_zcat(zbestfn, qn_flag, sq_flag, abs_flag, zcomb_flag)
  File "/global/common/software/desi/cori/desiconda/20200801-1.4.0-spec/code/desitarget/master/py/desitarget/lyazcat.py", line 141, in make_new_zcat
    zcat[col] = fms[zid][col]
ValueError: no field of name SV1_SCND_TARGET
```
This PR logs an error and leaves that column as zeros in the output file:
```
[cori06 ~] make_zqso_files -t 80605 -n 20210205 -o $CSCRATCH/desi/zqsotest
INFO:lyazcat.py:66:tmark: 
    Loading QuasarNP Model file and lines of interest: 2021-06-03 | 15:00:35
INFO:lyazcat.py:66:tmark: 
      QNP model file loaded: 2021-06-03 | 15:00:35
INFO:make_zqso_files:141:<module>: processing TILEID=80605, NIGHTID=20210205, petal=0
INFO:lyazcat.py:66:tmark: 
    Making redrock zcat: 2021-06-03 | 15:00:35
INFO:lyazcat.py:100:make_new_zcat: Read /global/cfs/cdirs/desi/spectro/redux/daily/tiles/cumulative/80605/20210205/zbest-0-80605-thru20210205.fits
ERROR:lyazcat.py:146:make_new_zcat: Input fibermap missing SV1_SCND_TARGET; leaving it blank
INFO:lyazcat.py:66:tmark: 
    Adding QuasarNP data: 2021-06-03 | 15:00:35
INFO:lyazcat.py:66:tmark: 
    Creating output file...: 2021-06-03 | 15:00:37
INFO:lyazcat.py:66:tmark: 
    --/global/cscratch1/sd/sjbailey/desi/zqsotest/tiles/cumulative/80605/20210205/zqso-0-80605-thru20210205.fits written out correctly.: 2021-06-03 | 15:00:37
...
```

It also records the quasarnp version used in a DEPNAMnn/DEPVERnn header.

@eleanorlyke and/or @geordie666, please confirm that leaving SV1_SCND_TARGET blank in the case of a missing input column is the correct behavior.

### CODE REVIEW

File: py/desitarget/lyazcat.py [hunk 1]
```diff
@@ -138,7 +140,10 @@ def make_new_zcat(zbestname, qn_flag=False, sq_flag=False, abs_flag=False,
     # ADM also add the appropriate bit-columns.
     Mxcols, _, _, = main_cmx_or_sv(fms, scnd=True)
     for col in Mxcols:
-        zcat[col] = fms[zid][col]
+        if col in fms.dtype.names:
+            zcat[col] = fms[zid][col]
+        else:
+            log.error(f'Input fibermap missing {col}; leaving it blank')
```
Comment (geordie666) [inline py/desitarget/lyazcat.py:146]: Consider, here, failing spectacularly with 
```
log.error(f'Input fibermap missing {col}; leaving it blank')
if col in zcatdatamodel.dtype.names:
    raise ValueError('{col} is required in the data model!')
```

**Review by geordie666 — COMMENTED** [review]
This looks simple enough. _But_ my preference is to check _which_ columns failed. I think we're fine allowing failures on columns that look like `SV1_*`, `SV2_*` and even `SV3_*` as we only started the `lyazcat` formalism with the Main Survey. But, we might want to die for some other missing columns?

I've tried to provide a code snippet that will complain if a missing column is truly _bad_ so that we don't have additional issues downstream in the MTL loop, by checking against the "formal" data model.

File: py/desitarget/lyazcat.py (hunk 1 above)
Comment (geordie666) [reply py/desitarget/lyazcat.py:146]: Actually, the only columns that are strictly needed for the MTL loop are:
```
from desitarget.mtl import zcatdatamodel as mtlzdm, survey_data_model
required_cols = survey_data_model(mtlzdm, survey="main").dtype.names
```
The other columns (such as the `DESI_/BGS_/MWS_/SCND_TARGET` columns and `SPECTYPE`) were added for "general utility" after a lengthy discussion about the `zqsocat` format on a  DESI data telecon. So, you might want to just check against `required_cols` for a spectacular fail.

**Comment by sbailey** [comment]
@geordie666 I added a check similar to what you suggested.  Unexpectedly missing optional columns are logged, while missing required columns trigger a ValueError.

I forgot to update master before branching so the doc/changes.rst is behind.  To avoid merge conflicts I will update it after this is approved and merged.

**Comment by geordie666** [comment]
@sbailey: This looks fine to me. Merge at will. Don't worry about updating the changes docs for this PR, I'll do that in my working branch.

**Pull request merged**